### PR TITLE
Budget/Transaction-Record App

### PR DESCRIPTION
**Pull request by Fatima-hub333**
### The following tasks are implemented here in this PR;
**Design**

- [ ]  You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:

- Colors.
- Typography: font face, size, and weight.
- Layout: composition and space between elements.

Original design idea by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

**Interactions**

- [ ] **Splash screen**

- A simple page with the name of your app (yes, you need to choose one), and links to the signup and log-in pages.

- [ ] **Sign up and log in pages**

- The user should be able to register in the app with their full name, email, and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- [ ] **Home page (categories page)**

- When the user logs in, they are presented with the categories page.
- For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- [ ] **Transactions page**

- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page, the user can see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

- [ ] **"Add a new category" page**
- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

- [ ] **"Add a new transaction" page**

- The user fills out a form to create a new transaction with:
- name (mandatory)
- amount (mandatory)
- categories (mandatory at least one)
- The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

**Testing requirements**
- Create unit and integration tests for all the most important components of your RoR application.

**Video Presentation**
[video](https://www.loom.com/share/2fc55c9260de4e2d8ca9ab946b5b3848)

**Additional Information:**
- Capybara is not compatible with windows laptops and as such, the integration test may not pass at your end
[See](https://questions.microverse.org/t/net-opentimeout-while-trying-to-test-capybara/3707/2)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,67 @@
-# Capstone-aka-Budget-App
-The BudgetApp is a mobile web application where you can manage your budget: you have a list of transactions associated with a category/many categories, so that you can see how much money you spent and on what item you spent it on
+# BudgetApp
```
Comment (GzimAsani) [inline README.md:1]: - [ ] Include a video presentation for your project, these are the requirements
image

File: README.md [hunk 2]
```diff
@@ -1,2 +1,67 @@
...
+
+> The BudgetApp is a mobile web application where you can manage your budget: you have a list of transactions associated with a category/many categories, so that you can see how much money you spent and on what item you spent it on
+---
+
+## Project Requirements
```
Comment (GzimAsani) [inline README.md:6]: - [ ] Add a live version of your project online

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<div>
```
Comment (GzimAsani) [inline app/views/devise/registrations/new.html.erb:1]: - [ ] Whenever a user writes a wrong email or password, you should flash a message indicating that
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @Fatima-hub333              :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your app, it looks very similar with the design,+1: 
 
 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 




 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (Fatima-hub333) [reply README.md:1]: Hi, sorry but vide presentation link is in PR . Can you double check it plz?

File: app/views/devise/registrations/new.html.erb (hunk 3 above)
Comment (Fatima-hub333) [reply app/views/devise/registrations/new.html.erb:1]: can you help me in this?

**Review by omar-muhamad — COMMENTED** [review]
# Invalid ❌

Hi @Fatima-hub333, @GzimAsani,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

Your Code Review Request is marked as invalid for these reasons:

- [ ] Please follow the correct Gitflow workflow because this the release 1.0 from the app so the Pull Request suppose to compare the development(dev) branch to the main(master) branch.

image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

File: app/views/devise/registrations/new.html.erb (hunk 3 above)
Comment (omar-muhamad) [reply app/views/devise/registrations/new.html.erb:1]: Hi @Fatima-hub333,
To do this requirement please check error handling partials rendered in this line 
https://github.com/Fatima-hub333/Capstone-aka-Budget-App/blob/2d0cec7bafbb60d428a58008e92dae6d4a98ef1e/app/views/devise/registrations/new.html.erb#L6
and this file too [app](https://github.com/Fatima-hub333/Capstone-aka-Budget-App/tree/featured/app)/[views](https://github.com/Fatima-hub333/Capstone-aka-Budget-App/tree/featured/app/views)/[layouts](https://github.com/Fatima-hub333/Capstone-aka-Budget-App/tree/featured/app/views/layouts)/application.html.erb and you can see this [video ](https://www.youtube.com/watch?v=RQQ98mbh2Ic)too

**Pull request merged**